### PR TITLE
manifest: SoftDevice Controller supporting coexistence and Periodic Advertising

### PR DESCRIPTION
**Pull request by rugeGerritsen**
Brings in new MPSL and SoftDevice Controller supporting:

- Bluetooth Coexistence
- Periodic advertising in the advertising state (experimental)

### CODE REVIEW

**Comment by rugeGerritsen** [comment]
Rebased on master to resolve conflict in manifest file

File: subsys/bluetooth/controller/hci_driver.c [hunk 1]
```diff
@@ -89,8 +89,21 @@ BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_PERIPHERAL) ||
 	#else
 		#define SDC_ADV_SET_COUNT 1
 	#endif
+	#define SDC_ADV_SET_MEM_SIZE \
+		(SDC_ADV_SET_COUNT * CONFIG_BT_CTLR_ADV_DATA_LEN_MAX)
```
Comment (seko-nordic) [inline subsys/bluetooth/controller/hci_driver.c:93]:
```suggestion
		(SDC_ADV_SET_COUNT * SDC_MEM_PER_ADV_SET(CONFIG_BT_CTLR_ADV_DATA_LEN_MAX))
```

Comment (rugeGerritsen) [reply subsys/bluetooth/controller/hci_driver.c]: Fixed

File: subsys/bluetooth/controller/hci_driver.c [hunk 2]
```diff
@@ -86,11 +86,26 @@ BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_PERIPHERAL) ||
...
 #else
 	#define SDC_ADV_SET_COUNT 0
+	#define SDC_ADV_SET_MEM_SIZE 0
+#endif
+
+#if defined(CONFIG_BT_PER_ADV)
+	#define SDC_PERIODIC_ADV_COUNT CONFIG_BT_EXT_ADV_MAX_ADV_SET
+	#define SDC_PERIODIC_ADV_MEM_SIZE \
+		(SDC_PERIODIC_ADV_COUNT * \
+		 SDC_MEM_PER_PERIODIC_ADV_SET(CONFIG_BT_CTLR_ADV_DATA_LEN_MAX))
```
Comment (seko-nordic) [inline subsys/bluetooth/controller/hci_driver.c:105]: should we use SDC_ADV_BUF_SIZE here as well?
Also I find it confusing when half of SDC_ defines comes from sdc.h and another half defined locally in this file. May be we should not use SDC_ prefix for local defines? It also 'breaks' namespace.

Comment (rugeGerritsen) [reply subsys/bluetooth/controller/hci_driver.c:105]: We could use `SDC_ADV_BUF_SIZE`, but when using periodic advertising `CONFIG_BT_CTLR_ADV_DATA_LEN_MAX` is always available. 

I agree that we should stop using the SDC prefix in this file. I will do that in a separate commit. Are you fine with doing that in a separate PR?

**Pull request merged**